elixir/test/samson_secret_puller_test.exs

```diff
@@ -0,0 +1,48 @@
+defmodule SamsonSecretPullerTest do
+  use ExUnit.Case
+
+  import ExUnit.CaptureIO
+
+  doctest SamsonSecretPuller
+
+  setup do
+    File.rm_rf!   "./test/secrets" # Make sure each test starts with a clean slate
+    File.mkdir_p! "./test/secrets"
+    File.write!   "./test/secrets/FOO", "bar"
+    {:ok, []}
+  end
+
+  test "reads secrets from the given folder" do
+    File.touch!   "./test/secrets/.done"
+
+    assert SamsonSecretPuller.fetch_secrets!("./test/secrets") == [FOO: "bar"]
```

Comment:
Nope, but the test framework fails in a much nicer way. It doesn't just say "assertion failed, no message given"

```
  1) test reads secrets from the given folder (SamsonSecretPullerTest)
     test/samson_secret_puller_test.exs:15
     Assertion with == failed
     code: [1, 2, 3] == [2, 3, 4]
     lhs:  [1, 2, 3]
     rhs:  [2, 3, 4]
     stacktrace:
       test/samson_secret_puller_test.exs:18
```